test/opt/spread_volatile_semantics_test.cpp

```diff
@@ -1139,155 +1139,135 @@ TEST_F(VolatileSpreadTest, NoInlinedfuncCalls) {
-%70 = OpAccessChain %_ptr_Function_float %rayDesc %int_1
-OpStore %70 %float_0
-%71 = OpAccessChain %_ptr_Function_float %rayDesc %int_3
-OpStore %71 %float_1000
-%72 = OpLoad %RayDesc %rayDesc
-%73 = OpCompositeExtract %v3float %72 0
-%74 = OpCompositeExtract %float %72 1
-%75 = OpCompositeExtract %v3float %72 2
-%76 = OpCompositeExtract %float %72 3
-%77 = OpLoad %Payload %myPayload_0
-OpStore %myPayload %77
-%78 = OpLoad %accelerationStructureNV %rs
+%param_var_count = OpVariable %_ptr_Function_uint Function
 ; CHECK: {{%\w+}} = OpLoad %uint %SubgroupSize Volatile
-%79 = OpLoad %uint %SubgroupSize
-OpTraceNV %78 %uint_0 %79 %uint_0 %uint_1 %uint_0 %73 %74 %75 %76 %uint_0
-%81 = OpLoad %Payload %myPayload
-OpStore %myPayload_0 %81
+%21 = OpLoad %uint %SubgroupSize
+OpStore %param_var_count %21
+%22 = OpFunctionCall %void %func3 %param_var_count
 OpReturn
 OpFunctionEnd
-%func1 = OpFunction %void DontInline %34
+%func1 = OpFunction %void DontInline %6
 %bb_entry_2 = OpLabel
 OpReturn
 OpFunctionEnd
+%func3 = OpFunction %void DontInline %25
+%count = OpFunctionParameter %_ptr_Function_uint
+%bb_entry_3 = OpLabel
+OpReturn
+OpFunctionEnd
 )";
   SinglePassRunAndMatch<SpreadVolatileSemantics>(text, true);
 }
 
+TEST_F(VolatileSpreadErrorTest, NoInlinedMultiEntryfuncCalls) {
+  const std::string text = R"(
+OpCapability RayTracingNV
+OpCapability SubgroupBallotKHR
+OpExtension "SPV_NV_ray_tracing"
+OpExtension "SPV_KHR_shader_ballot"
+%1 = OpExtInstImport "GLSL.std.450"
+OpMemoryModel Logical GLSL450
+OpEntryPoint RayGenerationNV %main "main" %SubgroupSize
+OpEntryPoint GLCompute %main2 "main2" %gl_LocalInvocationIndex %SubgroupSize
+OpSource HLSL 630
+OpName %main "main"
+OpName %bb_entry "bb.entry"
+OpName %param_var_count "param.var.count"
+OpName %main2 "main2"
+OpName %bb_entry_0 "bb.entry"
+OpName %param_var_count_0 "param.var.count"
+OpName %func "func"
+OpName %count "count"
+OpName %bb_entry_1 "bb.entry"
+OpDecorate %gl_LocalInvocationIndex BuiltIn LocalInvocationIndex
+OpDecorate %SubgroupSize BuiltIn SubgroupSize
+%uint = OpTypeInt 32 0
+%uint_0 = OpConstant %uint 0
+%_ptr_Input_uint = OpTypePointer Input %uint
+%float = OpTypeFloat 32
+%v4float = OpTypeVector %float 4
+%void = OpTypeVoid
+%12 = OpTypeFunction %void
+%_ptr_Function_uint = OpTypePointer Function %uint
+%_ptr_Function_v4float = OpTypePointer Function %v4float
+%29 = OpTypeFunction %void %_ptr_Function_v4float
+%34 = OpTypeFunction %void %_ptr_Function_uint
+%SubgroupSize = OpVariable %_ptr_Input_uint Input
+%gl_LocalInvocationIndex = OpVariable %_ptr_Input_uint Input
+%main = OpFunction %void None %12
+%bb_entry = OpLabel
+%param_var_count = OpVariable %_ptr_Function_uint Function
+%19 = OpLoad %uint %SubgroupSize
+OpStore %param_var_count %19
+%20 = OpFunctionCall %void %func %param_var_count
+OpReturn
+OpFunctionEnd
+%main2 = OpFunction %void None %12
+%bb_entry_0 = OpLabel
+%param_var_count_0 = OpVariable %_ptr_Function_uint Function
+%s2 = OpLoad %uint %SubgroupSize
```

Comment:
This is not exactly what I wanted.  Could you move the load and the other load of SubgroupSize into %func?  The idea is that you have 1 load in 1 function, but it is called from two different entry points.  This should fail with the same message.